quickstarts/04-llm-based-workflows/04_parallel_workflow.py

```diff
@@ -0,0 +1,163 @@
+import logging
+import time
+from typing import List
+
+import dapr.ext.workflow as wf
+from dapr.ext.workflow import DaprWorkflowContext
+from dotenv import load_dotenv
+from pydantic import BaseModel, Field
+
+from dapr_agents.llm.dapr import DaprChatClient
+from dapr_agents.workflow.decorators import llm_activity
+
+# Load environment variables (API keys, etc.)
+load_dotenv()
+
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+
+# Initialize the Dapr workflow runtime and LLM client
+runtime = wf.WorkflowRuntime()
+llm = DaprChatClient(component_name="openai")
+
+
+# ----- Models -----
+
+
+class Question(BaseModel):
+    """Represents a single research question."""
+
+    text: str = Field(..., description="A research question related to the topic.")
+
+
+class Questions(BaseModel):
+    """Encapsulates a list of research questions."""
+
+    questions: List[Question] = Field(
+        ..., description="A list of research questions generated for the topic."
+    )
+
+
+# ----- Workflow -----
+
+
+@runtime.workflow(name="research_workflow")
+def research_workflow(ctx: DaprWorkflowContext, topic: str):
+    """Defines a Dapr workflow for researching a given topic."""
+    # 1) Generate research questions
+    questions: Questions = yield ctx.call_activity(
+        generate_questions, input={"topic": topic}
+    )
+
+    # Handle both dict and model cases gracefully
+    q_list = (
+        [q["text"] for q in questions["questions"]]
+        if isinstance(questions, dict)
+        else [q.text for q in questions.questions]
+    )
```

Comment:
i see that we have to support both data forms often in our code. Do you by chance know why? Like in this quickstart can we just support one? I think it's bc the underlying state implementation supports just dictionaries, but I'm not sure if that is because of our implementation in dapr agents or because of limitations within the python sdk or something else tbh